tooling/ast_fuzzer/src/abi.rs

```diff
@@ -0,0 +1,85 @@
+use std::collections::BTreeMap;
+
+use noirc_abi::{Abi, AbiParameter, AbiReturnType, AbiType, AbiVisibility, Sign};
+use noirc_frontend::{
+    monomorphization::ast::{Program, Type},
+    shared::Visibility,
+};
+
+/// Generate the [Abi] interface of a [Program].
+pub fn program_abi(program: &Program) -> Abi {
+    let main = program.main();
+    let param_vis = program.main_function_signature.0.iter().map(|(_, _, vis)| vis);
+
+    let parameters = main
+        .parameters
+        .iter()
+        .zip(param_vis)
+        .map(|((_id, _is_mutable, name, typ), vis)| AbiParameter {
+            name: name.clone(),
+            typ: to_abi_type(typ),
+            visibility: to_abi_visibility(vis),
+        })
+        .collect();
+
+    let return_type = match &main.return_type {
+        Type::Unit => None,
+        typ => Some(AbiReturnType {
+            abi_type: to_abi_type(typ),
+            visibility: to_abi_visibility(&program.return_visibility),
+        }),
+    };
+
+    Abi { parameters, return_type, error_types: BTreeMap::default() }
+}
+
+/// Check if a type is valid as an ABI parameter for the `main` function.
+fn is_valid_in_abi(typ: &Type) -> bool {
+    match typ {
+        Type::Unit
```

Comment:
Just noting this is due to a bug https://github.com/noir-lang/noir/issues/5110 and we should accept unit as ABI input. We can link the issue here to track it